psp/models/recent_history.py

```diff
@@ -125,19 +125,24 @@ def __init__(
         *,
         pv_data_source: PvDataSource,
         nwp_data_source: NwpDataSource | None,
+        excarta_data_source: NwpDataSource | None,
         regressor: Regressor,
         random_state: np.random.RandomState | None = None,
         use_nwp: bool = True,
+        use_excarta: bool = True,
         nwp_variables: list[str] | None = None,
+        excarta_variables: list[str] | None = None,
         nwp_dropout: float = 0.0,
         pv_dropout: float = 0.0,
+        excarta_dropout: float = 0.0,
```

Comment:
Same idea as above, potentially change `nwp_dropout` to a list of floats of dropouts